### PR TITLE
[Bugfix][Spec Decode] Apply suppress_tokens on the Gemma 4 MTP sparse path

### PR DESCRIPTION
**Pull request by aminsamir45**
## Purpose

When `use_ordered_embeddings` is set on a Gemma 4 MTP drafter, `Gemma4Proposer._greedy_sample` samples through `get_top_tokens` rather than `compute_logits`:

```python
# vllm/v1/spec_decode/gemma4.py:113
return self.model.get_top_tokens(hidden_states)
```

That reaches `Gemma4MTPMaskedEmbedder.get_top_tokens`, which takes a sparse argmax directly over the centroid-selected candidates:

```python
logits, indices = self._select_and_score(hidden_states, lm_head_weight)
return indices.gather(-1, logits.argmax(-1, keepdim=True)).squeeze(-1)
```

`compute_logits` — the only place `suppress_tokens` is applied — is never called on this path, so **the drafter can propose tokens the generation config marks as suppressed**. The same applies to the CUDA-graph-captured variant, since `_setup_centroids_cuda_graphs` captures `masked_emb.get_top_tokens` directly.

This is a correctness gap rather than a crash, so it is silent: the suppressed token is proposed by the drafter and then either accepted or rejected by the verifier depending on the target model's own distribution.

## Approach

Give `Gemma4MTPMaskedEmbedder` a boolean mask over the vocabulary, populated from the draft model's generation config in `Gemma4MTP.__init__`, and apply it to the candidate logits before the argmax.

The mask lives on the embedder rather than on `Gemma4MTP` because there are two call sites — `Gemma4MTP.get_top_tokens` and the proposer's direct `masked_emb.get_top_tokens` during graph capture — and only module-level state covers both.

It is registered with `persistent=False`: it is derived from the generation config, not a checkpoint weight, and a persistent buffer would surface as an unexpected key during weight loading.

The dense path is untouched — `forward` still returns unsuppressed full-vocab logits and `compute_logits` masks them, which the test asserts explicitly.

## Test plan

Adds `tests/model_executor/test_gemma4_mtp_sparse_suppress.py` (CPU-only, `@pytest.mark.cpu_test`). `Gemma4MTPMaskedEmbedder` has no `VllmConfig` dependency, so these construct it for real with a deterministic centroid layout: an identity `token_ordering`, fixed centroid weights so the selected candidate set is always tokens 0–7, and an `lm_head_weight` giving an unambiguous ranking (token 5 > 2 > 6).

- suppressing the best candidate returns the second-best, and suppressing both returns the third;
- `[]` and `None` are no-ops;
- `suppress_mask` stays out of `state_dict` while `token_ordering` remains in it;
- suppression holds across a batched `hidden_states`;
- `forward()` output is byte-identical with and without suppression configured.

## Known degenerate case, deliberately not matched

If *every* centroid-selected candidate is suppressed, the sparse argmax returns an arbitrary suppressed candidate, whereas the dense path returns the lowest-id non-selected token — non-selected positions are filled with `finfo.min`, which compares above `-inf`. Reaching this requires `suppress_tokens` to cover every token in the selected clusters. I left the behavior rather than encoding a somewhat arbitrary dense-path artifact as a contract, but happy to match it if reviewers prefer.

## Relationship to existing work

#48693 by @dumko2001 also touches sparse-path parity as part of a broader Gemma 4 refactor; it has been in draft with merge conflicts since July. None of its code is used here. This is deliberately scoped to the sparse suppression gap alone so it can be reviewed independently.

Depends on nothing, but is adjacent to the separate CUDA-graph fix for `compute_logits` on the dense path (#48503).

## Validation caveat

No GPU access here, so this has not been exercised on a real Gemma 4 MTP run, and the test file has not been executed against a real vLLM install. What was verified: the assertions above all pass against the actual patched `Gemma4MTPMaskedEmbedder` class source executed standalone against real torch, and `ruff check` / `ruff format` (pinned v0.14.0) are clean. Hardware confirmation would be welcome.

## AI assistance disclosure

Parts of this change and its test were drafted with AI assistance (Claude). I have reviewed every changed line and validated the behavior described above.

### CODE REVIEW

**Comment by jonfairbanks** [comment]
Hey @aminsamir45, any chance you could rebase to help get this one merged? 🙏 I don't want to add a 3rd pending PR to the pile but would love to take advantage of this feature. Thanks!
